### PR TITLE
new: add a cron job in CI to automatically bump tables.

### PR DESCRIPTION
**Pull request by FedeDP**
The job will create a PR against the repo.

### CODE REVIEW

File: .github/workflows/update-tables.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          signoff: true
+          base: master
+          branch: update/tables
+          title: 'update: update syscalls tables.'
+          body: |
+            This autogenerated PR updates the tables of syscalls from kernel ${{ env.VERSION }}. Do not edit this PR.
+            ```
+          commit-message: 'update: update syscalls tables.'
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (FedeDP) [inline .github/workflows/update-tables.yml:48]: The github token must have write access to the repo.

File: .github/workflows/update-tables.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v5
+        with:
+          signoff: true
+          base: master
+          branch: update/tables
+          title: 'update: update syscalls tables.'
+          body: |
+            This autogenerated PR updates the tables of syscalls from kernel ${{ env.VERSION }}. Do not edit this PR.
```
Comment (FedeDP) [inline .github/workflows/update-tables.yml:45]: Put the kernel version in the PR body!

**Comment by FedeDP** [comment]
This can also be manually triggered since it has a workflow_dispatch trigger; this way, you can test it to check whether it works fine.

File: .github/workflows/update-tables.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Checkout syscalls-table ⤵️
+        uses: actions/checkout@v4
+        with:
+          ref: master
+      
+      - name: Checkout linux ⤵️
+        run: |
+          git clone https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git linux/ --depth=1
+          
+      - name: Fetch linux version
```
Comment (hrw) [inline .github/workflows/update-tables.yml:24]: "make kernelversion" is enough ;D

File: .github/workflows/update-tables.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name: Update Syscalls Tables
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * 1' # Each monday
```
Comment (hrw) [inline .github/workflows/update-tables.yml:6]: I prefer Tuesday as safer day. Nevermind which timezone Linus Torvalds will be RC will be already done.

**Review by hrw — COMMENTED** [review]
Nice.

I am doing changes in https://github.com/hrw/syscalls-table/tree/merge-with-python-one branch so this one has to wait a bit.

File: .github/workflows/update-tables.yml (hunk 3 above)
Comment (FedeDP) [reply .github/workflows/update-tables.yml:24]: Oh thank you!

**Comment by hrw** [comment]
Thanks!

**Pull request merged**